Automattic/woocommerce-payments
# Remove Task List Feature Flag and task list refactors

Fixes #4489 

#### Changes proposed in this Pull Request





* Removes the feature flag relating to the overview task list (as it is no longer used).
* I also took the opportunity to make a refactor which will make the TaskList logic simpler. This includes:
* Moving all tasks to their own files within a `tasks` folder. Previously, the implementation mixed business logic on the content of tasks with the display logic, this change compartmentalizes the logic of the TaskList and Task components.
* Converting each file to TypeScript, and some additional fixes/clarifications of types to help with this.





#### Testing instructions

Ideally, this PR should lead to no changes in functionality.

You can test it as follows:

**Prerequisites**

* Checkout `develop` and pull the latest version.
* Run `npm run listen` on your server.
* Create a new store (you can onboard a PO account in order to test the PO task).
* Place an order on your store using the test card `4000000000000259`. (which will create a dispute in the store).

**Testing on `develop`**

* Apply [this patch](https://gist.github.com/dmallory42/21d1f99964fe00a41ed66bb84a6a769d) to ensure the tasks always show.
* Go to the `Account > Overview page.
* You should see all the tasks (there will be 4 in total) - Check that each one has sensible copy, and that the actions are as expected. They should be as described:
    - **Update Business Details:** Will take the user to Stripe Express.
    - **Resolve Dispute** Will take the user to the active dispute.
    - **Reconnect WPCOM** This task should take the user to a link to reconnect WPCOM, however it will take you to a non-existent page because we are only simulating the task appearing (the conditions for it to appear are not valid).
    - **PO Verify Account Task** - Will take the user to Stripe KYC to finish onboarding.

**Testing on this branch**

* Checkout this branch and pull the latest version.
* Apply [this patch](https://gist.github.com/dmallory42/b45f0ac8f804697344fec5998dff3ecc) to ensure the tasks always show. (Note: this is a different patch from the first one!)
* Test each scenario from `develop` and ensure the behaviour is the same. (The tasks should look and feel the same.)
* Note that I made a few changes to make the tasks more consistent in terms of behaviour, for example all tasks are now always expanded and their action is tied to clicking on the action button. But overall behaviour should be consistent.

-------------------


**Post merge**
